src/sentry/api/endpoints/debug_files.py

```diff
@@ -101,8 +129,10 @@ def get(self, request, project):
         :auth: required
         """
         download_requested = request.GET.get("id") is not None
-        if download_requested and (request.access.has_scope("project:write")):
+        if download_requested and (has_download_permission(request, project)):
```

Comment:
Will checking only the role prevent organization access tokens from accessing this endpoint now? It seems like they previously would have been able to access this endpoint.

Reply:
Should we change the check to project:write or sufficient role?
That might cause some unexpected behavior when the minimum role would be set to the Owner (project:write is starting with admin).

Any suggestions here? I'm not sure how does it work. Maybe we could check for project:write when accessing with the organization access token, otherwise, check the role?

Reply:
It might not be a workflow that customers actually use and could be safe to ignore. Combining the scope and role check together will partially invalidate the role check and feel broken.

Reply:
@jan-auer is this something we need to worry about?

Reply:
It is potentially, yes. The `has_download_permission` function is essentially copied from [`EventAttachmentDetailsPermission`](https://github.com/getsentry/sentry/blob/da8591550a30c0285fdc6763b3d625f832076817/src/sentry/api/endpoints/event_attachment_details.py#L16). I believe at the time we did not even have organization access tokens, or at least for attachments it was less important. I would say if customers can use organization tokens to upload files, then they should also be able to download them with the same tokens again. 

@markstory I'm not 100% familiar with the latest state of authentication. My understanding is that we have:
- API Keys. They are soft-deprecated, and the Settings UI links to user-based API Tokens. Those use an `AnonymousUser`, which I cannot comprehend if it has access.
- API Tokens. They are user-based, and discouraged for automation. Since those are tied to the user, they will work.
- Sentry Integration tokens. Those create a proxy user. The corresponding App access returns all Teams, but the OrganizationMember query will fail. 

I think what is missing is the query for `SentryApp` that has been introduced about 5 months ago [here](https://github.com/getsentry/sentry/pull/18996/files#diff-1ea12c8a0f5ac7b426649197a032141aa0d0b9870857eab9fb306f6b2bb58649R51).